python/ray/remote_function.py

```diff
@@ -97,6 +99,13 @@ def _remote(self,
         """An experimental alternate way to submit remote functions."""
         worker = ray.worker.get_global_worker()
         worker.check_connected()
+
+        if self._last_export_time < worker._last_shutdown_time:
```

Comment:
This seems like a nice way to allow remote functions to be reused between multiple sessions without redefining them.

There are some moderately rare ways this could go wrong since `time.time()` is not monotone.

That could be fixed by using some sort of counter. E.g., the worker keeps some sort of `session_index` which increments every time you call `shutdown`. Then when a remote function is exported, we set its `session_index` equal to the worker's `session_index`.

I haven't thought through the details, but I'm not sure if this will work for nested remote functions.

Reply:
Good point.